src/persist/src/file.rs

```diff
@@ -237,8 +252,15 @@ impl FileBuffer {
         // unless that file was just created, in which case default to 0.
         let mut bytes = Vec::new();
         metadata_file.read_to_end(&mut bytes)?;
-        let seqno_start = if bytes.len() != 0 {
-            assert!(bytes.len() == 8);
+        let len = bytes.len();
+        let seqno_start = if len != 0 {
+            if len != 8 {
```

Comment:
nit: the `read_exact` call will return an error if there isn't 8 bytes, so this can probably be cleaned up by calling map_err there and substituting this one before the "?"

Reply:
thanks!